File: 027_remove_element.py
from typing import List
# // two pointer loop nums, if current element  != val, move it to curr pointer, increment curr, return curr after loop

# //time:O(n)
# //space:O(1)

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        curr = 0 

        for i in range(len(nums)): 
            if nums[i] != val:
                nums[curr] = nums[i] 
                curr += 1 
        return curr

File: 027_removeElement.cs
// two pointer loop nums, if current element  != val, move it to curr pointer, increment curr, return curr after loop

//time:O(n)
//space:O(1)

public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int curr = 0; 
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums[i] != val)
            {
                nums[curr] = nums[i]; 
                curr++;
            }
        }
        return curr;
    }
}